ksql-engine/src/test/java/io/confluent/ksql/planner/plan/JoinNodeTest.java

```diff
@@ -481,6 +483,44 @@ public void shouldNotPerformJoinIfInputPartitionsMisMatch() {
     assertEquals(JoinNode.JoinType.OUTER, joinNode.getJoinType());
   }
 
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldFailJoinIfTableCriteriaColumnIsNotKey() {
```

Comment:
Did you mean `join criteria column`?

Reply:
No I mean the table's column in the join criteria.